### ISSUE TITLE
custom function for tickformat

### ISSUE THREAD

**Issue by monuminu** [plotly/plotly.py#2693]
Is there anyway i can give a custom function as tickformat ?

layout=go.Layout(
    showlegend=True, 
    legend=dict(x=1.05,y=1.1),
    xaxis=dict(showticklabels=True),
    yaxis=dict(
                tickformat=my_function_name,
                showgrid=False,
))

**Comment by nicolaskruchten** [comment]
Unfortunately not. Figures are always serialized to JSON, meaning we cannot accept functions for any argument.

**Comment by Sulstice** [comment]
Yeah this is probably one of my biggest pain points in using plotly, maybe can we get more detailed docs on proper formatting? Since you guys implement the string formatting route ex: ".2f%" reading through those docs are extremely painful and time consuming.

**Comment by monuminu** [comment]
for example in our organization millions are shown as MM instead of M . How can i solve the problem . Could you please help ?

**Comment by nicolaskruchten** [comment]
The string formatting directives we use come from the d3-format library and are documented here: https://github.com/d3/d3-format

Unfortunately, I don't know of any directive to render millions as "MM" :(

**Comment by maddenfederico** [comment]
I'd like to bump this. Even if we can't get custom functions for formatting, I'd like there to be some way to format ticklabels that are plain strings, at the very least to control things like character width, etc. Would it not be possible to implement something similar to hovertemplates?

**Comment by gvwilson** [comment]
Hi - we are trying to tidy up the stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for several years, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. Thanks for your help - @gvwilson
